src/resolve/resolved-dns-synthesize.c

```diff
@@ -321,17 +326,17 @@ static int synthesize_gateway_rr(Manager *m, const DnsResourceKey *key, int ifin
 
         af = dns_type_to_af(key->type);
         if (af >= 0) {
-                n = local_gateways(m->rtnl, ifindex, af, &addresses);
+                n = lookup(m->rtnl, ifindex, af, &addresses);
```

Comment:
shouldn't there be an assert on `lookup`? (maybe not, getting late and might be confused)

Reply:
yeah, we should add that. done